### PR TITLE
compilation: fixed compilation by clang 3.6

### PR DESCRIPTION
**Pull request by hoopoepg**
fix for issue https://github.com/openucx/ucx/issues/1784

### CODE REVIEW

**Comment by hoopoepg** [comment]
@alinask could you look at this?
thank you

File: test/gtest/common/gtest-all.cc [hunk 1]
```diff
@@ -7921,7 +7921,7 @@ const char kCurrentDirectoryString[] = ".\\";
 # endif  // GTEST_OS_WINDOWS_MOBILE
 #else
 const char kPathSeparator = '/';
-const char kPathSeparatorString[] = "/";
+//const char kPathSeparatorString[] = "/";
```
Comment (MattBBaker) [inline test/gtest/common/gtest-all.cc]: Should this be removed instead of commented out?

**Comment by MattBBaker** [comment]
okay to test

**Comment by mellanox-github** [comment]
Can one of the admins verify this patch?

File: test/gtest/uct/test_p2p_err.cc [hunk 2]
```diff
@@ -118,7 +118,8 @@ class uct_p2p_err_test : public uct_p2p_test {
     static void* get_unused_address(size_t length)
     {
         void *address = NULL;
-        ucs_status_t status = ucs_mmap_alloc(&length, &address, 0, "test_dummy");
+        ucs_status_t status = ucs_mmap_alloc(&length, &address, 0
+                UCS_MEMTRACK_NAME("test_dummy"));
```
[the remark this answers is not in the document]
Comment (hoopoepg) [reply test/gtest/uct/test_p2p_err.cc]: fixed

File: test/gtest/common/gtest-all.cc (hunk 1 above)
Comment (hoopoepg) [reply test/gtest/common/gtest-all.cc]: removed
thank you

**Comment by shamisp** [comment]
@hoopoepg - thanks for the patch !!!

Do you have signed UCX contributor agreement ? 

If not, can you please sign the agreement and send a copy to ORNL ?
Corporate:
http://www.openucx.org/wp-content/uploads/2016/01/ucx-corporate-contributor-agrement-v1.3.pdf
Or Individual:
http://www.openucx.org/wp-content/uploads/2016/01/ucx-individual-contributor-agrement-v1.3.pdf

**Comment by hoopoepg** [comment]
@shamisp I updated profile on GitHub

**Comment by alinask** [comment]
bot:retest

**Comment by hoopoepg** [comment]
bot:bgate:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4393/ for details (Mellanox internal link).

**Comment by hoopoepg** [comment]
bot:bgate:retest

**Comment by hoopoepg** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4404/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4413/ for details (Mellanox internal link).

**Comment by hoopoepg** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4445/ for details (Mellanox internal link).

**Comment by hoopoepg** [comment]
bot:retest

**Comment by miked-mellanox** [comment]
consistently breaks on arm

**Comment by hoopoepg** [comment]
right, every time on new place in statistics... will look what is wrong (completely forgot about ARM :) )

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4454/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4458/ for details (Mellanox internal link).

**Comment by hoopoepg** [comment]
bot:mlx:retest

**Comment by alinask** [comment]
@MattBBaker Can you please post the ORNL failure?

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4461/ for details (Mellanox internal link).

**Comment by MattBBaker** [comment]
bot:ornl:retest

**Comment by miked-mellanox** [comment]
@hoopoepg - please open a ticket to add clang test into contrib/test_jenkins.sh
Thanks

**Pull request merged**